### PR TITLE
chore(ci): release calert helm charts

### LINKED ISSUE #48: Publish the chart to a helm registry / ArtifactHub
We'd like to deploy calert using the helm terraform provider.
This provider requires "built" helm charts, published within a helm registry.
Publishing the chart can be done within a github action and targeted to the gh-pages branch which can be exposed as a Github pages directory. ref: https://github.com/marketplace/actions/helm-chart-releaser

### PR DESCRIPTION
**Pull request by aslafy-z**
Closes #48 
Closes #50

Before merging this, you'll have to
- create an empty `gh-pages` branch by following this gist: https://gist.github.com/ramnathv/2227408#creating-a-clean-gh-pages-branch

- enable github pages from your repository settings and choose the `gh-pages` branch option

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -33,3 +33,26 @@ jobs:
           args: release --rm-dist
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  release:
```
Comment (aslafy-z) [inline .github/workflows/release.yml:37]:
```suggestion
  helm-chart:
```

**Comment by mr-karan** [comment]
Hi @aslafy-z Thanks for this. Let me know when it's ready to be reviewed!

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -33,3 +33,28 @@ jobs:
...
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
+
+      - name: Run chart-releaser
```
Comment (aslafy-z) [inline .github/workflows/release.yml:55]:
```suggestion
      - name: Bump chart release
        shell: |
          echo

      - name: Run chart-releaser
```

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -33,3 +33,33 @@ jobs:
...
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Install Helm
+        uses: azure/setup-helm@v3
```
Comment (aslafy-z) [inline .github/workflows/release.yml:53]:
```suggestion
        uses: azure/setup-helm@v3
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by aslafy-z** [comment]
@mr-karan This looks good to me. What do you think about syncing `calert` versions and `calert` chart versions?

**Comment by mr-karan** [comment]
Might be related (?) https://github.com/mr-karan/calert/issues/56

**Comment by mr-karan** [comment]
@aslafy-z I missed this, sorry! A bit unclear, do you mean the helm chart version and calert version? I guess we can keep them same?

**Comment by granescb** [comment]
@mr-karan  Sorry, will be progress here? We really wanna use this chart on our pipeline

**Comment by mr-karan** [comment]
Feel free to contribute, I don't really have any experience with helm.

**Comment by aslafy-z** [comment]
#56 is related, yes. This pr will allow the use of this repository GitHub pages as a helm repository. The current implementation maps the git release tag to the helm version. @granescb can review to the impl so we can go forward?

**Comment by aslafy-z** [comment]
Up @mr-karan. Did you find some time to review this pr?

**Comment by mr-karan** [comment]
@aslafy-z Sorry for late response. Yes, LGTM. We can use the git tag release version for Helm charts. I'm merging this.

**Comment by joaogbcravo** [comment]
Hey!

I'm getting a 404 error here:

```
helm repo add calert https://mr-karan.github.io/calert/charts
helm repo update
Error: looks like "https://mr-karan.github.io/calert/charts" is not a valid chart repository or cannot be reached: failed to fetch https://mr-karan.github.io/calert/charts/index.yaml : 404 Not Found
 ```

@aslafy-z enumerated some requirements on the [first post](https://github.com/mr-karan/calert/pull/50#issue-1349501569). Is that missing?

**Comment by aslafy-z** [comment]
It is indeed @joaogbcravo 
@mr-karan can you please enable GH pages as explained in op?

**Pull request merged**